src/Mod/PartDesign/Gui/Command.cpp

```diff
@@ -1804,23 +1815,32 @@ void CmdPartDesignThickness::activated(int iMsg)
     Q_UNUSED(iMsg);
     Gui::SelectionObject selected;
     bool useAllEdges = false;
-    if (!dressupGetSelected ( this, "Thickness", selected, useAllEdges))
+    bool noSelection = false;
+    if (!dressupGetSelected ( this, "Thickness", selected, useAllEdges, noSelection))
         return;
 
-    Part::Feature *base = static_cast<Part::Feature*>(selected.getObject());
-    std::vector<std::string> SubNames = std::vector<std::string>(selected.getSubNames());
-    size_t i = 0;
-
-    // filter out the edges
-    while(i < SubNames.size())
-    {
-        std::string aSubName = static_cast<std::string>(SubNames.at(i));
 
-        if (aSubName.compare(0, 4, "Face") != 0) {
-            // empty name or any other sub-element
-            SubNames.erase(SubNames.begin()+i);
+    Part::Feature* base;
+    std::vector<std::string> SubNames;
+    if (noSelection) {
+        base = static_cast<Part::Feature*>(PartDesignGui::getBody(true)->Tip.getValue());
+    }
+    else {
+        base = static_cast<Part::Feature*>(selected.getObject());
+        SubNames = std::vector<std::string>(selected.getSubNames());
+
+        // filter out the edges
+        size_t i = 0;
+        while (i < SubNames.size())
+        {
+            std::string aSubName = static_cast<std::string>(SubNames.at(i));
```

Comment:
I know it's not really your code, but why is this `static_cast` here? Doesn't `at()` give you a std:string??

Reply:
Yes it does return a std::string. I don't know why this was here.